lang/perl/lib/Avro/DataFileReader.pm

```diff
@@ -133,12 +133,12 @@ sub next {
 
     my @objs;
 
-    $datafile->read_block_header if $datafile->eob;
+    $datafile->read_block_header if $datafile->eob and not $datafile->eof;
     return ()                    if $datafile->eof;
```

Comment:
Why don't you move this line before line 136 ?

Reply:
Good point.